### PR TITLE
feat: generate-sitemap workflow

### LINKED ISSUE #12460: Move all sitemaps / rss feeds into a better workflow
# Chore

## Describe the chore

Perhaps move the RSS feed and Sitemap generation into a GitHub action.

Currently what is happening is .xml files are appearing in git commits as they are being generated on `npm run build` locally.

### PR DESCRIPTION
**Pull request by brandongregoryscott**
## What kind of change does this PR introduce?

Resolves #12460

Feature: This PR adds a new Github Actions workflow file for running the `apps/docs` and `apps/www` builds (which generate updated sitemap files in their respective `postbuild` npm scripts). These files are then committed and pushed up to a branch (named `generate-sitemap`, but can be changed) and PRed into `master` by the Github Actions bot. This should automate the workflow a bit and keep the sitemap up-to-date without a ton of manual intervention.

## What is the current behavior?

Currently, the sitemap files are generated on local build, committed, and PRed in manually (https://github.com/supabase/supabase/pull/12463 as an example)

## What is the new behavior?

The workflow added in this PR will watch for pushes to the `master` branch, run a build in the `apps/docs` and `apps/www` directories, commit any changed .xml files, and create or update a PR for the sitemap. The PR will still need to be manually merged in by someone, but it's less effort than doing the rest manually. We can probably configure it to auto-merge, too.

## Additional context

Since the sitemap files will now be handled in CI, it might make sense to remove the calls to `generate-sitemap.mjs` in `postbuild` scripts and instead let the workflow explicitly call it. Otherwise, people will still probably accidentally commit and push these changes from local development. I've added a separate `build:sitemap` script in the `apps/www` package.json file for this purpose.

### CODE REVIEW

File: .github/workflows/generate-sitemap.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+            apps/www/**/*.xml
+            apps/docs/**/*.xml
+          base:
+            master
+          title:
+            Generate Sitemap
+          branch:
+            generate-sitemap
+          commit-message:
+            'docs: generate sitemap'
```
Comment (MildTomato) [inline .github/workflows/generate-sitemap.yml:54]:
```suggestion
'www / docs: generate sitemap'
```

**Comment by MildTomato** [comment]
@brandongregoryscott wdyt about checking for changes in /apps/docs and /apps/www first before doing any builds?
Will help needlessly running the builds on commits that don't involve those sites.

**Comment by brandongregoryscott** [comment]
> @brandongregoryscott wdyt about checking for changes in /apps/docs and /apps/www first before doing any builds? Will help needlessly running the builds on commits that don't involve those sites.

Yeah, makes sense! Tested it out on my fork and pushed it up in this commit: [`9ef05f1` (#12514)](https://github.com/supabase/supabase/pull/12514/commits/9ef05f14bbf5bc3b6e6e49feb5e460e2b87c68c8)

**Comment by MildTomato** [comment]
@joshenlim @alaister 

can you folks check this PR?

**Comment by alaister** [comment]
Hey @brandongregoryscott,

Sorry for the delay on this one too.

We decided that sitemaps shouldn't be included in the repo and should be created on build every time. So I'm going to close this – thanks for all your work here, though!

If you do want to continue this, we'll happily accept another PR that gitignores the sitemaps from the repo.

Thanks again 🙌🏻

**Pull request closed without merging**